src/jolt/gui/jolt_app.py

```diff
@@ -168,13 +183,78 @@ def __init__(self, simulated=TEST_NOHW):
             logging.debug("Changing front-end offset from %s to %s", old_fe_offset, fe_offset)
             self.dev.set_frontend_offset(fe_offset)
 
+        # Override XRC default values, since they are static and do not work for different JOLT versions
+        self.spinctrl_voltage.SetMin(0)  # For now, now differentiation for the lower bound
+        self.spinctrl_voltage.SetMax(self.dev.voltage_range[1])
+        self.voltage_gui = min(self.voltage, self.dev.voltage_range[1])
+        self.spinctrl_voltage.SetValue(self.voltage_gui)
+
         # Start the thread for polling
         self.polling_thread = threading.Thread(target=self.do_poll)
         self.polling_thread.start()
 
+        # Initialize calibration thread
+        self.calibration_thread = threading.Thread(target=self.calibrate)
+
+        # Calibration is only available for JOLT V2
+        self.valid_calibration = False
+
+        if self.dev.version == 2:
+            # If no manually selected calibration was found, pick the latest one available on the system
```

Comment:
Please move (almost) all the content of this if block into  a separate function (which load the calibration file). The only thing that should stay is the message dialog in case the calibration failed to be loaded.